android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt

```diff
@@ -1182,6 +1215,34 @@ class HmssdkFlutterPlugin :
         }
     }
 
+    private val keyChangeListener = object : HMSKeyChangeListener {
+        override fun onKeyChanged(key: String, value: String?) {
+            val args = HashMap<String, Any?>()
+            args["event_name"] = "on_key_changed"
+            val keyValuePair = HashMap<String,String?>()
+            keyValuePair["key"] = key
+            keyValuePair["value"] = value
+            args["data"] = keyValuePair
+            CoroutineScope(Dispatchers.Main).launch {
+                sessionStoreSink?.success(args)
+            }
+        }
+    }
+    private fun addKeyChangeListener(call: MethodCall,result: Result){
+
+        val keys = call.argument<List<String>>("keys") ?: run {
+            HMSErrorLogger.returnArgumentsError("keys parameter is null")
+        }
+
+        keys.let { keys as List<String>
```

Comment:
The optional  ` ? ` check for keys is not added here `keys.let`. 
But it is added in the `HMSSessionStoreAction.kt` class below in the `setSessionMetadataForKey` function.
Which one is the intended way?

Reply:
made changes here.